### ISSUE TITLE
METRIC_OUTPUT_DIR not available.

### ISSUE THREAD

**Issue by hyoda** [microsoft/nni#3177]
**Environment**:
- NNI version: 1.9
- NNI mode (local|remote|pai): local
- Client OS: Windows
- Server OS (for remote mode only): Windows
- Python version: 3.6
- PyTorch/TensorFlow version: None
- Is conda/virtualenv/venv used?: Yes
- Is running in Docker?: Nope.

**Log message**:
 - nnimanager.log: 

[2020-12-10 8:22:29 PM] INFO [ 'Stopping local machine training service...' ]
[2020-12-10 8:22:29 PM] ERROR [ 'GPU scheduler error: ChildProcessError: Command failed: cmd.exe /c taskkill /PID ﻿10660 /T /F\nERROR: The process "﻿10660" not found.\r\n `cmd.exe /c taskkill /PID ﻿10660 /T /F` (exited with error code 128)' ]
[2020-12-10 8:22:29 PM] INFO [ 'Change NNIManager status from: STOPPING to: STOPPED' ]

 - dispatcher.log:
 - nnictl stdout and stderr:
 

**What issue meet, what's expected?**:
$env:METRIC_OUTPUT_DIR ="C:\Users\hyoda\AppData\Local\Temp/hyoda/nni/script"

![image](https://user-images.githubusercontent.com/9294616/101769611-8fabd380-3b2a-11eb-8305-8d3f979aa2af.png)

Yes, path is not available.

I know where it is.

/**
 * GPUScheduler for local training service
 */
class GPUScheduler {

    private gpuSummary!: GPUSummary;
    private stopping: boolean;
    private readonly log: Logger;
    private readonly gpuMetricCollectorScriptFolder: string;

    constructor() {
        this.stopping = false;
        this.log = getLogger();
        **this.gpuMetricCollectorScriptFolder = `${os.tmpdir()}/${os.userInfo().username}/nni/script`;**

**How to reproduce it?**: 

**Additional information**:

If it is possible I would like to fix it. But I can't please, fix that. Thank you.

**Comment by SparkSnail** [comment]
Do you mean that `C:\Users\hyoda\AppData\Local\Temp/hyoda/nni/script` does not exist in your machine? Could you please run your experiment with `--debug` mode, and paste nniManager.log here?

**Comment by hyoda** [comment]
Sorry. I apologize for not explaining in detail.
I mean, that path really exists.
However, when reading the path from the source code, the path is specified differently due to the OS (in case of MS WINDOWS, the identifier indicating the path ("/" => "\")).
```

So in the gpu_metrics_collector.ps1 file(In C:\Users\hyoda\AppData\Local\Temp\hyoda\nniscript) it looks like this:

$env:METRIC_OUTPUT_DIR="C:\Users\hyoda\AppData\Local\Temp**/hyoda/nni/script**"
$app = Start-Process "python" -ArgumentList "-m nni_gpu_tool.gpu_metrics_collector" -passthru -NoNewWindow
Write $app.ID | Out-File C:\Users\hyoda\AppData\Local\Temp\hyoda\nni\script\pid -NoNewline -encoding utf8

So when actually running a process (PID), it will eventually fail.
```

If it's normal, it should look like this:
`
$env:METRIC_OUTPUT_DIR="C:\Users\hyoda\AppData\Local\Temp\hyoda\nniscript"`

Of course, looking at the source code that creates the file, "path.join" is used, but it doesn't seem to work properly.
```
export function runGpuMetricsCollector(scriptFolder: string): void {
     if (process.platform ==='win32') (
         const scriptPath = path.join(scriptFolder,'gpu_metrics_collector.ps1');
         const content = String.Format(GPU_INFO_COLLECTOR_FORMAT_WINDOWS, scriptFolder, path.join(scriptFolder,'pid'));
         fs.writeFile(scriptPath, content, {encoding:'utf8' }, () => {runScript(scriptPath); });
     } else {
         cp.exec(getGpuMetricsCollectorBashScriptContent(scriptFolder), {shell:'/bin/bash' });
     }
}
```


What about using "path.join" in the constructor?
```
/**
 * GPUScheduler for local training service
 */
class GPUScheduler {

    private gpuSummary!: GPUSummary;
    private stopping: boolean;
    private readonly log: Logger;
    private readonly gpuMetricCollectorScriptFolder: string;

    constructor() {
        this.stopping = false;
        this.log = getLogger();
        this.gpuMetricCollectorScriptFolder = `${os.tmpdir()}/${os.userInfo().username}/nni/script`;
    }
```
Thank you so much.

**Comment by SparkSnail** [comment]
Yes, seems a bug in NNI. By the way, I noticed there is ` 'GPU scheduler error: ChildProcessError: Command failed: cmd.exe /c taskkill /PID ﻿10660 /T /F\nERROR: The process "﻿10660" not found.\r\n cmd.exe /c taskkill /PID ﻿10660 /T /F (exited with error code 128)' ` error in your log file, seems the pid 10660 does not exist in your machine, does your experiment running successful? Is there gpu_metric file under `C:\Users\hyoda\AppData\Local\Temp/hyoda/nni/script`?

**Comment by hyoda** [comment]
Thank you for confirming. Yes. There was a file in that path.
A month ago, there were traces of what was normally recorded in the "gpu_metrics" file.
I don't know exactly why, but it seems that it was temporarily activated while analyzing the nni code.

It was through the log that I became aware of this problem. It didn't work in nni. So when I ran the executable file (gpu_metrics_collector.ps1) in that path via "cmd", it failed.
When I corrected the path and made it run, it started to be recorded.

Thank you.

![image](https://user-images.githubusercontent.com/9294616/102408641-40393c00-4031-11eb-8dbf-1cddfb64a3e9.png)

![image](https://user-images.githubusercontent.com/9294616/102408676-492a0d80-4031-11eb-8dc8-aa202b6978f6.png)
